notebook/tests/selenium/utils.py

```diff
@@ -222,6 +222,14 @@ def set_cell_metadata(self, index, key, value):
         JS = 'Jupyter.notebook.get_cell({}).metadata.{} = {}'.format(index, key, value)
         return self.browser.execute_script(JS)
 
+    def get_cells_mode(self):
+        JS = 'return Jupyter.notebook.get_cells().map(function(c) {return c.mode;})'
+        return self.browser.execute_script(JS)
+
+    def get_cells_length(self):
```

Comment:
Not sure it's worth adding this for one call, when code can do `len(notebook)`.

Reply:
Solved the issue with the fccc7d7 commit